### PR TITLE
[RDY] plugin: enable the matchit plugin by default

### PR DESCRIPTION
**Pull request by fmoralesc**
As per #2676 .

### CODE REVIEW

**Comment by splinterofchaos** [comment]
LGTM :+1:

**Comment by fmoralesc** [comment]
I force pushed a fix for a typo (May doesn't have 219 days, as far I remember, and if it did, today is not the 219th): https://github.com/neovim/neovim/pull/2723/files#diff-cf85c69b12fa07d87121cbe9313746b3R3

**Comment by fmoralesc** [comment]
Marking as RDY, since I don't think more changes will be needed for this.

File: runtime/doc/matchit.txt [hunk 1]
```diff
@@ -18,7 +11,6 @@ For Vim version 6.3.  Last change:  2007 Aug 29
 5. Known Bugs and Limitations				|matchit-bugs|
 
 The functionality mentioned here is a plugin, see |add-plugin|.
-This plugin is only available if 'compatible' is not set.
 You can avoid loading this plugin by setting the "loaded_matchit" variable
 in your |vimrc| file: >
 	:let loaded_matchit = 1
```
Comment (Pyrohh) [inline runtime/doc/matchit.txt]: This file's modeline needs to be updated (I just took the one in `nvim_intro.txt`).

**Comment by Pyrohh** [comment]
There are many broken links throughout `matchit.txt`, which should probably be fixed given `doc/` currently only has one broken link (in one of the `pi_*.txt` files, forget which one). `make html` should show them.

File: runtime/doc/matchit.txt (hunk 1 above)
Comment (Pyrohh) [inline runtime/doc/matchit.txt]: Also, large parts of the `*matchit-activate*` section are now redundant, as is line 43 given what's been happening to Vi references.

File: runtime/doc/usr_05.txt [hunk 2]
```diff
@@ -399,45 +399,26 @@ Further reading:
...
-you already have the directory.) >
-
-	:!mkdir ~/.vim
-	:!mkdir ~/.vim/plugin
-	:!cp $VIMRUNTIME/macros/matchit.vim ~/.vim/plugin
-
-The "cp" command is for Unix, on MS-DOS you can use "copy".
-
-Now create a "doc" directory in one of the directories in 'runtimepath'. >
+   Let suppose a plugin ("ns") comes with a help file in a non-standard place.
```
Comment (Pyrohh) [inline runtime/doc/usr_05.txt]: `Let` -> `Let us`? Also maybe something longer as an example name: how about `my-plugin`?

**Comment by fmoralesc** [comment]
Thanks, @Pyrohh, will fix those things.

**Comment by fmoralesc** [comment]
Marked as RFC again, in case I missed something else (like that would ever happen, ha!).

File: runtime/doc/pi_paren.txt [hunk 3]
```diff
@@ -51,10 +51,9 @@ are:
   buffer-local equivalents b:matchparen_timeout and
   b:matchparen_insert_timeout.
 
-If you would like the |%| command to work better, the matchit plugin can be
-used, see |matchit-install|.  This plugin also helps to skip matches in
-comments.  This is unrelated to the matchparen highlighting, they use a
-different mechanism.
+If you would like the |%| command to work better, the |matchit| plugin can be
+used.  This plugin also helps to skip matches in comments.  This is unrelated
+to the matchparen highlighting, they use a different mechanism.
```
Comment (Pyrohh) [inline runtime/doc/pi_paren.txt]: `matchparen` has a help item, so I think surrounding `|` would be good here.

That's all I have for now (I'll take another look tomorrow, I just mentioned this so I wouldn't forget).

Comment (fmoralesc) [inline runtime/doc/pi_paren.txt]: Hm... this is literally in the documentation file for matchparent, so that link would just take you to the top.

Comment (Pyrohh) [inline runtime/doc/pi_paren.txt]: ah, didn't notice that.
it's fine with me i suppose.

**Comment by fmoralesc** [comment]
I'll squash if none else has some suggestions.

File: runtime/doc/usr_05.txt [hunk 4]
```diff
@@ -399,45 +399,27 @@ Further reading:
...
-	:!mkdir ~/.vim
-	:!mkdir ~/.vim/plugin
-	:!cp $VIMRUNTIME/macros/matchit.vim ~/.vim/plugin
-
-The "cp" command is for Unix, on MS-DOS you can use "copy".
-
-Now create a "doc" directory in one of the directories in 'runtimepath'. >
+   Let us suppose a plugin ("my-plugin"), which comes with a help file in a
+non-standard place (it usually resides in a sub-folder called `doc/`).
+   First, create a "doc" directory in one of the directories in 'runtimepath'. >
```
Comment (Pyrohh) [inline runtime/doc/usr_05.txt]: I don't think this should be indented. It makes it look like it's the start of another paragraph, even though this sentence is directly related to the previous.

File: runtime/doc/pi_paren.txt (hunk 3 above)
Comment (Pyrohh) [inline runtime/doc/pi_paren.txt]: in any case, shouldn't the comma in `highlighting, ...` instead be a colon or em-dash?

File: runtime/doc/matchit.txt [hunk 5]
```diff
@@ -267,7 +236,7 @@ have only one group; the effect is undefined if a group has only one pattern.
 A simple example is >
 	:let b:match_words = '\<if\>:\<endif\>,'
 		\ . '\<while\>:\<continue\>:\<break\>:\<endwhile\>'
-(In Vim regular expressions, |\<| and |\>| denote word boundaries.  Thus "if"
+(In Vim regular expressions, \< and \> denote word boundaries.  Thus "if"
```
Comment (Pyrohh) [inline runtime/doc/matchit.txt]: These two seem to have tags, but they have a forward slash in front of them:
```
							*/\<*
\<	Matches the beginning of a word: The next char is the first char of a
	word.  The 'iskeyword' option specifies what is a word character.
	|/zero-width|

							*/\>*
\>	Matches the end of a word: The previous char is the last char of a
	word.  The 'iskeyword' option specifies what is a word character.
	|/zero-width|

```

File: runtime/doc/matchit.txt [hunk 6]
```diff
@@ -307,8 +276,8 @@ if keywords are only recognized after the start of a line or after a
 semicolon (;), with optional white space.
 
 					*matchit-backref* *matchit-\1*
-In any group, the expressions |\1|, |\2|, ..., |\9| refer to parts of the
-INITIAL pattern enclosed in |\(|escaped parentheses|\)|.  These are referred
+In any group, the expressions \1, \2, ..., \9 (see |/\1|) refer to parts of the
```
Comment (Pyrohh) [inline runtime/doc/matchit.txt]: :+1:

File: runtime/doc/matchit.txt [hunk 7]
```diff
@@ -307,8 +276,8 @@ if keywords are only recognized after the start of a line or after a
...
+INITIAL pattern enclosed in escaped parentheses.  These are referred
 to as back references, or backrefs.  For example, >
```
Comment (Pyrohh) [inline runtime/doc/matchit.txt]: maybe quotes around the two mentioned terms?

File: runtime/doc/motion.txt [hunk 8]
```diff
@@ -1159,7 +1159,7 @@ remembered.
 					cursor is on the # or no ([{
 					following)
 			For other items the matchit plugin can be used, see
-			|matchit-install|.  This plugin also helps to skip
+			|matchit|.  This plugin also helps to skip
 			matches in comments.
```
Comment (Pyrohh) [inline runtime/doc/motion.txt]: this sentence could probably be reflowed

**Comment by Pyrohh** [comment]
Besides the nits, LGTM.

**Comment by fmoralesc** [comment]
Will fix the points @Pyrohh points, and then squash tomorrow if none has more to say.

File: runtime/doc/usr_05.txt (hunk 4 above)
Comment (fmoralesc) [inline runtime/doc/usr_05.txt]: I think it is meant to be another paragraph; it's the way it was formatted in the original, didn't want to change it that much.

**Comment by Pyrohh** [comment]
And please feel free to mark as RDY as well. I'll take another look soon.

**Comment by Pyrohh** [comment]
Make sure to rebase too :-)

**Comment by fwalch** [comment]
@Pyrohh I think this has been rebased, want to take another look?

**Comment by fmoralesc** [comment]
Yes, it is rebased, sorry for not notifying of it ;)

File: runtime/doc/matchit.txt [hunk 9]
```diff
@@ -150,35 +142,12 @@ setting |b:match_skip|.
...
-Either way, the script should start working the next time you start up Vim.
-
-(Earlier versions of the script did nothing unless a |buffer-variable| named
-|b:match_words| was defined.  Even earlier versions contained autocommands
-that set this variable for various file types.  Now, |b:match_words| is
-defined in many of the default |filetype-plugin|s instead.)
-
 For a new language, you can add autocommands to the script or to your vimrc
 file, but the recommended method is to add a line such as >
 	let b:match_words = '\<foo\>:\<bar\>'
```
Comment (Pyrohh) [inline runtime/doc/matchit.txt]: Do you see any reason to mention the non-recommended method at all?

Comment (fmoralesc) [inline runtime/doc/matchit.txt]: Not really, I'll change it to this:
~~~
For a new language, you can add a line such as >
	let b:match_words = '\<foo\>:\<bar\>'
to the corresponding |filetype-plugin|.  See |b:match_words| below for how
this variable is interpreted.
~~~

**Comment by Pyrohh** [comment]
LGTM so I merged. Thanks for reminding me @fwalch, and thanks @fmoralesc.

**Pull request closed without merging**